[DOC-313] Added a catalog flag - ysql_yb_toast_catcache_threshold (#22113)
yugabyte/yugabyte-db: docs/content/preview/reference/configuration/yb-tserver.md

Default: `true`

##### ysql_yb_toast_catcache_threshold

> Specifies the threshold beyond which catalog tuples will get compressed when storing in the PostgreSQL catalog cache. Setting this flag reduces memory usage for certain large objects including, functions and views in exchange for slower catalog refreshes.

Reviewer: ```suggestion
Specifies the threshold (in bytes) beyond which catalog tuples will get compressed when they are stored in the PostgreSQL catalog cache. Setting this flag reduces memory usage for certain large objects, including functions and views, in exchange for slower catalog refreshes.

```

> To enable this flag, it is recommended to set it at 2KB or higher, in order to minimize performance impact.

Reviewer: ```suggestion
To minimize performance impact when enabling this flag, set it to 2KB or higher.
```

> Default: -1 (disabled), minimum: 128 bytes

Reviewer: ```suggestion
Default: -1 (disabled). Minimum: 128 bytes.
```

Default: `true`

> ##### ysql_yb_toast_catcache_threshold

Reviewer: I'm assuming bytes, even though KB is mentioned
